src/main/scheduler/scheduler.c

```diff
@@ -286,7 +286,7 @@ FAST_CODE timeUs_t schedulerExecuteTask(task_t *selectedTask, timeUs_t currentTi
         float period = currentTimeUs - selectedTask->lastExecutedAtUs;
 #endif
         selectedTask->lastExecutedAtUs = currentTimeUs;
-        selectedTask->lastDesiredAt += (cmpTimeUs(currentTimeUs, selectedTask->lastDesiredAt) / selectedTask->desiredPeriodUs) * selectedTask->desiredPeriodUs;
+        selectedTask->lastDesiredAt += cmpTimeUs(currentTimeUs, selectedTask->lastDesiredAt);
```

Comment:
This is incorrect - multiplication is not the inverse of integer division.

Reply:
(4/2)*2 = 4 
Where incorrect ? Please provide your calcs..

Reply:
Are these the only values that you tested? In the future can you please test changes you propose properly before you open a pull request?

`(5 / 2) * 2 = 4`